### ISSUE TITLE
compilation failed for package 'nvimcom' (Windows 10, Gvim 8.0.3)

### ISSUE THREAD

**Issue by danielcalencar** [jalvesaq/Nvim-R#321]
Dear all, 

I cannot get Nvim-R working. I installed the plugin using `pathogen' (i.e., git clone in the bundle directory), then I tried opening an R file and typing \rf. I got an error message saying that the compilation of nvimcom have failed. Then, I came across some issue reports here, so I tried to build the nvimcom package manually (with no success).

I got the following messages:

```
$ R CMD INSTALL nvimcom_0.9-70.tar.gz
In R CMD INSTALL
* installing to library 'C:/Users/danie/OneDrive/Documents/R/win-library/3.4'
* installing *source* package 'nvimcom' ...
** libs
c:/Rtools/mingw_32/bin/gcc  -I"C:/R/R-3.4.4/include" -DNDEBUG       -D`C:/R/R-3.4.4/bin/R --slave -e 'cat(Sys.info()[[1]])'` -DLWS_NO_FORK  -DWIN32   -O3 -Wall  -std=gnu99 -mtune=generic -c nvimcom.c -o nvimcom.o
The filename, directory name, or volume label syntax is incorrect.
<command-line>:0:1: error: macro names must be identifiers
make: *** [nvimcom.o] Error 1
Warning: running command 'make -f "Makevars.win" -f "C:/R/R-3.4.4/etc/i386/Makeconf" -f "C:/R/R-3.4.4/share/make/winshlib.mk" SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)' SHLIB="nvimcom.dll" OBJECTS="nvimcom.o"' had status 2
ERROR: compilation failed for package 'nvimcom'
* removing 'C:/Users/danie/OneDrive/Documents/R/win-library/3.4/nvimcom'
* restoring previous 'C:/Users/danie/OneDrive/Documents/R/win-library/3.4/nvimcom'
```
I am using the following tools:

- Rtools 3.5, 32-bit only (Installed in C:\Rtools)
- R 3.4.4, 32-bit only (Installed in C:\R\R-3.4.4\)

When installing Rtools 3.5, I chose both the options of installing the utilities (make etc.) and not installing the utilities (I also have cygwin installed). And none of these options worked out.

I don't know what else I could try. Would you have some suggestions for me?

I am really excited to use this plugin with Gvim!

Regards, Daniel

**Comment by jalvesaq** [comment]
Does your .Rprofile outputs anything? If it does, gcc on Windows gets confused.
Some people have problem with RTools 3.4, but reported success with Rtools 3.3...

**Comment by danielcalencar** [comment]
Hi jalvesaq,

Thanks for your prompt reply!

I didn't see any .Rprofile being generated. It would have been generated in the directory that my getwd() outputs right?

Also, I tried installing Rtools 3.3 and R 3.3.3 (since I noticed that Rtools 3.3 is not compatible with R 3.4.4 (which I was using before).

Now I get this message:

```
$ R CMD INSTALL nvimcom_0.9-70.tar.gz
* installing to library 'C:/R/R-3.3.3/library'
* installing *source* package 'nvimcom' ...
** libs
c:/Rtools/mingw_32/bin/gcc  -I"C:/R/R-3.3.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"  -D`C:/R/R-3.3.3/bin/R --slave -e 'cat(Sys.info()[[1]])'` -DLWS_NO_FORK  -DWIN32   -O3 -Wall  -std=gnu99 -mtune=core2 -c nvimcom.c -o nvimcom.o
The filename, directory name, or volume label syntax is incorrect.
<command-line>:0:1: error: macro names must be identifiers
make: *** [nvimcom.o] Error 1
Warning: running command 'make -f "Makevars.win" -f "C:/R/R-3.3.3/etc/i386/Makeconf" -f "C:/R/R-3.3.3/share/make/winshlib.mk" SHLIB="nvimcom.dll" OBJECTS="nvimcom.o"' had status 2
ERROR: compilation failed for package 'nvimcom'
* removing 'C:/R/R-3.3.3/library/nvimcom'
```

**Comment by jalvesaq** [comment]
> I didn't see any .Rprofile being generated.
> It would have been generated in the directory that my getwd() outputs right?

Yes, if you had `getwd()` or any other code outputting any text it would corrupt the output of `cat(Sys.info()[[1]])`.

> The filename, directory name, or volume label syntax is incorrect.
> `<command-line>:0:1:` error: macro names must be identifiers

I'm sorry, but I don't know how to fix this error... Although I have Windows in my laptop (dual boot), I only use Linux. Perhaps someone else who uses Windows could have any clue as why gcc is failing in your case.

**Comment by danielcalencar** [comment]
Hi jalvesaq,

So I just tried open an .R file and run the \rf command. And it worked! I don't actually know why trying to build it manually didn't work

I just have another question now. I get the following when typing \rf 

![nvim-r](https://user-images.githubusercontent.com/3970011/38770689-f45e300a-3fe4-11e8-9fdb-3416a23f71b0.JPG)

Is there a way to make it open in an additional Gvim buffer instead. More like in the following?

![nvim-r2](https://user-images.githubusercontent.com/3970011/38770714-5c100f20-3fe5-11e8-8f43-78cb03cad16b.JPG)

Thanks for your help!

Regards

**Comment by danielcalencar** [comment]
Hi jalvesaq,

I think I found the answer for my question with the let `R_in_buffer=1` option.

Anyways, I think we can close this issue report

Thanks again,

Daniel

**Comment by jalvesaq** [comment]
The command to build `nvimcom` includes the `--no-multiarch` argument.

**Comment by BarbaraRobson** [comment]
I'm running into the same problem (Also using Windows 10  and I'm using R 3.6.0).
After some investigation, I found that Makevar.win is sending 'cat(Sys.info()[[1]])' as input to R including the quotation marks, so R just echos the string instead of evaluating it. Editing out the quotation marks produced a different error, so I edited it to replace the call to R with what should be the output (i.e. "Windows"). Makevar.win now looks like this:
``#PKG_CFLAGS=-D`${R_HOME}/bin/R --slave -e 'cat(Sys.info()[[1]])'` -DLWS_NO_FORK  -DWIN32``
`PKG_CFLAGS=-DWindows -DLWS_NO_FORK  -DWIN32`
`PKG_LIBS=-lWs2_32`
`.PHONY: all nvimapps`
`all: $(SHLIB) nvimapps`
`$(SHLIB):`
`nvimapps:`
`	(cd apps; make -f Makefile.win)`
That got me successfully past that line, produce nvimcom.o, and on to the next part:
`(cd apps;make -f Makefile)`
Unfortunately, I am now getting Make error 127. Still trying to figure out why.
